flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointITCase.java

```diff
@@ -321,7 +321,7 @@ public void invoke(Long value, Context context) throws Exception {
 			if (backpressure) {
 				// induce backpressure until enough checkpoints have been written
 				if (random.nextInt(100) == 42) {
-					Thread.sleep(0, 100_000);
+					Thread.sleep(100);
```

Comment:
Simplify to just `Thread.sleep(1)`?

Reply:
With modified `Thread.sleep()` I presume CPU usage went down, right?

But what about test duration? Has it increased? If so maybe 
```
if (random.nextInt(100) == 42) {
	Thread.sleep(1);
}
```
is better? 

Note that previous `Thread.sleep(0, 100_000);` was basically an equivalent of `Thread.sleep(0);` which in turn was burning 100% cpu. In other words
```
if (random() % 100 == 0) {
  Thread.sleep(1);
}
```
works usually much better and more like you would be expecting compared to
```
Thread.sleep(0, 100_000);
```
which is for the most purposes just broken and not sleeping at all.

Reply:
Test duration was lower altogether because of the source throttling. In the worst case, with this change, a checkpoint needs 100+100 ms compared to the ideal time of 100 ms. But since the test is much more resource friendly now, it was effectively lower on my 8-core machine.

The issue with your proposition is now that there may be runs without any backpressure, which may or may not be beneficial.

Reply:
Err, I meant to say that two instances of backpressure may not occur that I like to see covered:
- Backpressure results in full buffers to be snapshotted and restored. If we just accumulate 100 record, then with the current record size and the min buffer size (4kb), it's likely that only a few bytes are snapshotted and restored and also only individual buffers and not multiples. So, we may never see in the ITCase if multiple buffers are snapshotted/recovered at all and in the correct order.
- Consequently, we may also not see backpressure in all exchanges, but just the last exchange. But I think it's important to cover all exchanges to test the different kinds of recovery, especially during rescaling (to come). It's also good to cover backpressure being propagated to the source ultimately.